sale_product_approval/models/sale_order.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2021 Open Source Integrators
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from datetime import date
+
+from odoo import SUPERUSER_ID, _, fields, models
+from odoo.exceptions import UserError
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    approved_sale = fields.Boolean(
+        related="product_id.sale_ok",
+        string="Approved for Sale",
+        store=True,
+        default=True,
+    )
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    exceptions_sale_approval = fields.Boolean(
+        compute="_compute_exceptions",
+        string="Exception",
+    )
+
+    def _compute_exceptions(self):
```

Comment:
Missing dependencies
```suggestion
    @api.depends("order_line.approved_sale")
    def _compute_exceptions(self):
```